sdk/examples/xo_python/sawtooth_xo/processor/xo_state.py

```diff
@@ -0,0 +1,198 @@
+                        game.split(',')
+                        for game in state_data.decode().split('|')
+                    ]
+                }
+
+                board, state, player1, player2 = game_list[self._game_name]
+
+            except ValueError:
+                raise InternalError("Failed to deserialize game data.")
+
+        else:
+            game_list = {}
+            board = state = player1 = player2 = None
+
+        self._game = game_list
+        self._board = board
+        self._state = state
+        self._player1 = player1
+        self._player2 = player2
+
+    def delete_game(self):
+        del self._game[self._game_name]
+        if self._game:
+            self._store_game()
+        else:
+            self._delete_game()
+
+    def _store_game(self):
+        state_data = '|'.join(sorted([
+            ','.join([name, board, state, player1, player2])
+            for name, (board, state, player1, player2) in self._game.items()
+        ])).encode()
+
+        addresses = self._context.set_state(
+            {self._game_address: state_data},
+            timeout=self.TIMEOUT)
+
+        if len(addresses) < 1:
+            raise InternalError("State set error: {}".format(
+                self._game_address))
+
+    def _delete_game(self):
+        addresses = self._context.delete_state(
+            [self._game_address],
+            timeout=self.TIMEOUT)
+
+        if len(addresses) < 1:
+            raise InternalError("State delete error: {}".format(
+                self._game_address))
+
+    def store_game(self):
+        """Store the game state in the validator state.
+        """
+
+        self._store_game()
+
+    def create_game(self):
+        """Create an empty board at the beginning of play.
+        """
+
+        self._game[self._game_name] = ('-' * 9, "P1-NEXT", '', '')
+
+    def update_board(self, board):
+        """Update the game state with new board state.
+
+        Args:
+            board (str): 9 characters representing the xo board.
+        """
+
+        _, state, player1, player2 = self._game[self._game_name]
+
+        self._game[self._game_name] = (board, state, player1, player2)
+
+    def update_state(self, state):
+        """Update the variable tracking which player's turn it is.
+
+        Args:
+            state (str): The state of the game. e.g. P1-NEXT, P2-WIN
+        """
+
+        board, _, player1, player2 = self._game[self._game_name]
+
+        self._game[self._game_name] = (board, state, player1, player2)
+
+    def create_player1(self, player1):
+
+        board, state, _, player2 = self._game[self._game_name]
+
+        self._game[self._game_name] = (board, state, player1, player2)
+
+    def create_player2(self, player2):
+
+        board, state, player1, _ = self._game[self._game_name]
+
+        self._game[self._game_name] = (board, state, player1, player2)
+
+    @property
+    def state(self):
```

Comment:
So as not confuse users, can we call this game state or something?  Calling a field state in a class called XoState is moderately confusing.

Reply:
Yeah, agreed.